### DISCUSSION TITLE
[help] Is it possible to get tab autocomplete functionality when working with targets?

### DISCUSSION THREAD

**Discussion by matiasandina** [ropensci/targets#1205] [Help]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

This might be a problem of how Rstudio finds and autocompletes variables, but I would like to know if it's possible to find existing targets when attempting to read or to modify them on other targets. Some examples:

You try to read a target named (lineplots), but forgot the exact name:

```
tar_read(lineplot) <- autocomplete will not help you, tar_read will obviously fail
Error:
! target lineplot not found
```

You try to write a new target 

```
 tar_target(
    lineplots,
    ...
  ),
  # a bunch of targets here
   tar_target(
   modif_lineplot, 
   command = lineplot + theme_bw() # this will also fail, and autocomplete will not find lineplots
  )
```

I might need to work on better names, but I find it hard to remember the name of the functions and the targets themselves with no autocomplete. If anything, I find myself typing, doing CTRL+ F,  and scrolling through code way more than I'm used to.

**Comment by wlandau** [comment]
Auto-complete in the general case is tricky. Something might be possible through an RStudio addin like https://github.com/ropensci/targets/blob/main/R/rstudio_addin_tar_read.R. `targets` has plenty of addins, and one like this would be in scope. I'm not sure how to implement one for auto-complete, but I would be happy to review a pull request for this.

**Reply by wlandau** [reply]
A list of possible target names could come from tar_meta()$name, and you could cache it in memory so `targets` does not reread from `_targets/meta/meta` unless the time stamp of the file changed. Then it's a matter of using those possible target names for RStudio-powered auto-complete magic, if it is possible.

**Reply by wlandau** [reply]
Please feel free to submit a pull request if you think of something more feasible.
